diff --git a/geogrid/src/parallel_module.F b/geogrid/src/parallel_module.F
--- a/geogrid/src/parallel_module.F
+++ b/geogrid/src/parallel_module.F
@@ -1033,12 +1033,17 @@
 
       ! Local variables
 #ifdef _MPI
       integer :: mpi_ierr, mpi_errcode
 
-      call MPI_Abort(MPI_COMM_WORLD, mpi_errcode, mpi_ierr)
+      if (nprocs > 1) then
+         mpi_errcode = 1
+         call MPI_Abort(MPI_COMM_WORLD, mpi_errcode, mpi_ierr)
+      else
+         call MPI_Finalize(mpi_ierr)
+      end if
 #endif
 
-      stop
+      stop 1
 
    end subroutine parallel_abort
  
diff --git a/metgrid/src/parallel_module.F b/metgrid/src/parallel_module.F
--- a/metgrid/src/parallel_module.F
+++ b/metgrid/src/parallel_module.F
@@ -1033,12 +1033,17 @@
 
       ! Local variables
 #ifdef _MPI
       integer :: mpi_ierr, mpi_errcode
 
-      call MPI_Abort(MPI_COMM_WORLD, mpi_errcode, mpi_ierr)
+      if (nprocs > 1) then
+         mpi_errcode = 1
+         call MPI_Abort(MPI_COMM_WORLD, mpi_errcode, mpi_ierr)
+      else
+         call MPI_Finalize(mpi_ierr)
+      end if
 #endif
 
-      stop
+      stop 1
 
    end subroutine parallel_abort
  
